lib/puppet/provider/package/pip.rb

```diff
@@ -56,7 +56,13 @@ def query
   # cache of PyPI's package list so this operation will always have to
   # ask the web service.
   def latest
-    client = XMLRPC::Client.new2("http://pypi.python.org/pypi")
+    proxy = ENV["http_proxy"]
+    if proxy.nil? then
+      client = XMLRPC::Client.new2("http://pypi.python.org/pypi")
+    else
+      host, port = proxy[7..-1].split(':')
+      client = XMLRPC::Client.new2("http://pypi.python.org/pypi", proxy_host=host, proxy_port=port)
```

Comment:
As such, this could simply become `client = XMLRPC::Client.new2("http://pypi.python.org/pypi", ENV['http_proxy'] || ENV['HTTP_PROXY'])` without having to split or check for existence.